csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -116,49 +116,49 @@ internal static class NativeMethods
         #region SessionOptions API
 
         [DllImport(nativeLib, CharSet = charSet)]
-        public static extern IntPtr /*OrtSessionOptions* */ OrtCreateSessionOptions();
+        public static extern IntPtr /*(OrtStatus*)*/ OrtCreateSessionOptions(out IntPtr /*(OrtSessionOptions**)*/ sessionOptions);
```

Comment:
Why does it could fail?

Reply:
Remember that the user of the API doesn't know that this is simply doing a 'new' of a simple POD behind the scenes. If the user gets a nullptr back, he would like to examine what went wrong. Additionally, for him/her some ORT APIs return a status code and some don't. For consistency and uniformity returning a Status in all APIs makes sense.

Reply:
First, if we use Windows or Linux system APIs as an example, many of them do never fail. 
Second. It may not work as you think. If it failed because of OOM, it may still return nullptr, indicate it was successful, but the newly created object is nullptr.

Reply:
Because the Env has been deleted, or the pointer is bad.. we can have many ways any of these APIs can fail if we decide to check for these cases.